src/UI/Implementation/Component/Layout/Renderer.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts.and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Layout;
+
+use ILIAS\UI\Implementation\Render\AbstractComponentRenderer;
+use ILIAS\UI\Renderer as RendererInterface;
+use ILIAS\UI\Component;
+
+class Renderer extends AbstractComponentRenderer {
+    /**
+     * @inheritdoc
+     */
+    public function render(Component\Component $component, RendererInterface $default_renderer) {
+        $this->checkComponent($component);
+
+        if ($component instanceof Component\Layout\Page) {
+            return $this->renderPage($component, $default_renderer);
+        }
+        if ($component instanceof Component\Layout\Sidebar) {
+            return $this->renderSidebar($component, $default_renderer);
+        }
+        if ($component instanceof Component\Layout\Metabar) {
+            return $this->renderMetabar($component, $default_renderer);
+        }
+    }
+
+    protected function renderSidebar(Component\Layout\Sidebar $component, RendererInterface $default_renderer) {
+        $tpl = $this->getTemplate("tpl.sidebar.html", true, true);
+        $entry_signal = $component->getEntryClickSignal();
+        $counter = 0;
+        foreach ($component->getEntries() as $entry) {
```

Comment:
Ich würde vorschlagen, den $counter nicht zu nutzen sondern stattdessen `$entry` in `$index => $entry` zu verwandeln und dann $index zu nutzen. Das scheint mir robuster.